lib/server/auth.js

```diff
@@ -114,48 +115,75 @@ class DataAccess extends EventEmitter {
     });
     LOGGER.debug('ACCESS GROUP IDS: ' + agroupIds.join(' '));
 
-    let authurl = config.provide().get('authurl');
+    let auth_server = config.provide().get('auth_server');
+    let authurl = auth_server.url;
     let reqbody = {groups: agroupIds};
-    let reqopts = url.parse(authurl);
-    reqopts.method = 'POST';
+    let reqopts = {};
+    reqopts.url     = url.parse(authurl);
+    reqopts.method  = 'POST';
+    reqopts.headers = {
+      'Content-Type': 'application/json'
+    };
+
+    if ( authurl.startsWith('https://') ) {
+      let auth_cert = auth_server.client_cert;
+      if ( auth_cert ) {
+        reqopts.cert = auth_cert;
+      }
+      let auth_key = auth_server.client_key;
+      if ( auth_key ) {
+        reqopts.key = auth_key;
+        let auth_passphrase = auth_server.client_key_passphrase;
+        if ( auth_passphrase ) {
+          reqopts.passphrase = auth_passphrase;
+        }
+      }
+      let auth_ca = auth_server.ca;
+      if ( auth_ca ) {
+        reqopts.ca = auth_ca;
+      }
+    }
+
     if (this.authtype === constants.AUTH_TYPE_TOKEN) {
       reqbody.token = identifier.trim();
-      reqopts.path = constants.AUTH_URL_TOKEN;
+      reqopts.url.path = constants.AUTH_URL_TOKEN;
     } else if (this.authtype === constants.AUTH_TYPE_USER) {
       reqbody.user = this._getName(identifier);
-      reqopts.path = constants.AUTH_URL_USER;
+      reqopts.url.path = constants.AUTH_URL_USER;
     }
     if (!(reqbody.token || reqbody.user)) {
       return this._authErr(new Error(`Invalid identifier "${identifier}"`));
     }
-    // TODO: this only works when sentry is running http. In future, we would like
-    // sentry to run https, so we will need to enable https requests here before then.
-    let req = http.request(reqopts, ( response ) => {
-      let body = '';
-      response.on('error', this._authErr);
-      response.on('data', (chunk) => {
-        body += chunk;
-      });
-      response.on('end', () => {
-        if (response.statusCode !== 200) {
+
+    reqopts.body = JSON.stringify(reqbody);
+
+    request(reqopts, ( error, response, body ) => {
+      if ( error ) {
+        this._authErr();
+      } else {
+        if ( response.statusCode !== 200 ) {
           this._authErr(new Error(
             `received ${response.statusCode} from authorisation server`));
-        }
-        let decision;
-        try {
-          decision = JSON.parse(body);
+        } else {
+          let decision;
+          try {
+            decision = JSON.parse(body);
+          } catch (e) {
+            this._authErr(new Error('Server response was invalid JSON'));
+          }
+          if ( !( 'ok' in decision ) ) {
```

Comment:
nit: why not `if (typeof decision.ok !== 'boolean')` ? Makes error clearer if server sends `ok: "true"` or something equally odd as an attempt to authorise. Or
```js
if (decision.ok === true) {
  ...
} else if (decision.ok === false) {
  ...
} else {
  this._authErr(new Error('Auth server response does not match expected format'));
}
```